src/components/application_manager/src/reset_timeout_handler_impl.cc

```diff
@@ -91,8 +85,13 @@ void ResetTimeoutHandlerImpl::on_event(const event_engine::Event& event) {
       auto request = it->second;
       if (static_cast<hmi_apis::FunctionID::eType>(request.hmi_function_id_) ==
           method_name) {
-        application_manager_.updateRequestTimeout(
-            request.connection_key_, request.mob_correlation_id_, timeout);
+        if (application_manager_.IsUpdateRequestTimeoutRequired(
+                request.connection_key_,
+                request.mob_correlation_id_,
+                timeout)) {
+          application_manager_.updateRequestTimeout(
```

Comment:
@LitvinenkoIra this proxy functions are redundant, please remove. Use directly request controller

Reply:
@LuxoftAKutsan I need app_manager here because ResetTimeoutHandler needs access to event_dispatcher 
https://github.com/smartdevicelink/sdl_core/pull/2748/files/501fe87a6ee8ff6ec63e214fdaded439f0a8169c#diff-df723fbe4c190b36a8b7862ee84cf663R44

And application_manager_.updateRequestTimeout(..) proxy function is used in many places, not only in ResetTimeoutHandler. 
So the problem is only in _IsUpdateRequestTimeoutRequired_ function. I believe that it's ok to leave it as it is.

Reply:
@LitvinenkoIra please always think about SOLID principles. 
If you have ability to avoid high coupling of components - avoid it. 
Please provide only required components to the ResetTimeoutHandler. 
*Proxy functions* in application manager is not a good practice. Remove this functions. In places where this functions used, replace it. If component have access to app_manager, it is also have access to RequestControler.

Reply:
@LitvinenkoIra yep. add getter for request controller.